### DISCUSSION TITLE
Keeping paths in same position on rerender

### DISCUSSION THREAD

**Discussion by RococoCoding** [xyflow/xyflow#1191] [Layouting]
I have a graph where you can click a + icon to insert a new node, but sometimes when the graph rerenders, the paths get switched up. So if I inserted a node under the delay on the right side, when the graph rerenders with the new node, that whole path on the right would be moved to the middle. It usually happens with middle/right paths. Is there a way to ensure the paths stay in the same order after inserting a new node?
![Screen Shot 2021-05-12 at 6 04 12 PM](https://user-images.githubusercontent.com/63833718/118049942-7f538b00-b34c-11eb-8378-fcb63894f46b.png)

Edit: Seems like this is a dagre issue that's been unresolved for a while.  https://github.com/dagrejs/dagre/pull/263

**Comment by visualsitemaps** [comment]
This sounds terrible. Such an obvious use-case.
 Is there any other solution to this than waiting for that dagre issue to be resolved?

**Comment by moklick** [comment]
Hey @RococoCoding 

nice graph :) If it's an unresolved dagre issue could you try to do the layouting with [elk](https://github.com/kieler/elkjs) for example? I used it before and it works really well (just the API is a bit weird).

**Reply by RococoCoding** [reply]
I'll check it out. Thanks for the tip!

**Comment by michaelpward** [comment]
Does anyone have an implementation (elk or otherwise) that solves this?

Having the same problem with paths shifting to entirely new locations when a node is added. It's a show-stopper for us - and we were so excited to find react-flow.

**Reply by moklick** [reply]
Hey. This seems to be an issue with dagre and not with react flow. You can try other layouting libraries like elk or cytoscape.
